aldus17

aleksander-GD opened the pull request.

aleksander-GD committed:
Added txt file with text

aleksander-GD committed:
Added the exercise from today's lecture with body styled from CSS.

aleksander-GD committed:
updated html and css files

aleksander-GD committed:
javascript exercise 1 with regex from lesson 2 added.
css exercise 1 from lesson 2 added.

aleksander-GD committed:
css exercise 1 done. advertisement banners needs improvement.

aleksander-GD committed:
removed script from html + script folder. Not part of css exercise 1.

aleksander-GD committed:
cat and mouse update.

aleksander-GD committed:
update to css exercise 1 and javascript mouse and cat.

aleksander-GD committed:
css e2 update

aleksander-GD committed:
Merge branch 'master' into aldus17

aleksander-GD committed:
exercise 1 and 2 php and ajax

aleksander-GD committed:
Assignment 1 update

aleksander-GD committed:
Assignment 1 update

aleksander-GD committed:
Assignment 1 update

styled pages and made others ready for styling. tried to add login, but needs to be fixed.

aleksander-GD committed:
Assignment 1 update

Added upload, tested ok.
Added migration.sql file.
Fixed major issue with include_once / require_once, solution was NOT TO TOUCH IT. 
TODO: feedpage, userlist (possible ajax), and of coures the most time consuming one, CSS.

aleksander-GD committed:
Assignment 1 update

cleaned up

aleksander-GD committed:
Assignment 1 update

Updated upload functionality
Imagefeed done
Userlist done.
TODO: security, ajax and proper regex check on registration form.

aleksander-GD committed:
Assignment 1 update

Javascript controls the attribute title and pattern for regex. The only thing missing on this part is the server side check on all the information.

aleksander-GD committed:
Assignment 1 update

Minor fixes
Added logout for every page
Ajax not working fully, still needs work. 
CSS styling for scaling needs to be done.

aleksander-GD committed:
Assignment 1 update

Finished CSS styling on all pages. 
Small minor bug fixes in regex form.
Changed default input file button with customized one. 
TODO: Ajax call

aleksander-GD committed:
Assignment 1 update

AJAX call done, makes a get request call. getUserlist.php fetches all records, stores them in an array and encodes them as Json, which then is echoed back to the userlist_page.php. This json data is decoded and afterwards is is appended to the HTML table.

aleksander-GD committed:
Assignment 1 update

index now shows an error if username or password is wrong.

aleksander-GD committed:
Migrations.sql updated

aleksander-GD committed:
Assignment 1 update

Ajax call added for search
Ajax call for userlist modified.
Cleaned up a bit.

nc-hela committed:
Exercises for week 2-5

aleksander-GD committed:
Assignment 1 update (requirements completed)

Cleaned up the code.
Made a sessionCheck() to check whether the sessions has started and sessionRedirect() to check if a user tries to access the pages without a login.
Updated migrations file

aleksander-GD committed:
Assignment 1 done, minor fixes and clean up

Minor fixes
Cleaned up in the code
Removed unnecessary functions

aleksander-GD committed:
deleted text file

JacobGlumby1 reviewed (approved):
**Bruger oplevelsen:**
Siden fungere i det store hele som den skal. Der er visuelt feedback på knapperne, man får kommentarer når man prøver at oprette en ulovlig bruger og man kan ikke navigere rundt på de forskellige sider uden at være logget ind. 
Der er elementer af 'responssivenes' - det går dog galt på image feed siden, når en kommentar til et billede bliver for lang. Der kunne i din CSS fil bruges en "overflow-wrap" på kommentaren til at bryde sætningen op i forhold til 'parent' containeren. 


** CSS:**

Det er fedt med opdeling af de forskellige ark til de forskellige sider. 
Måske man kunne have ét CSS ark til at style top menuen på alle de sider som har menuen, istedet for at duplikere dette i de forskellige ark. 

**Koden**
Godt med brug af OOP frem for 'procedural' programmering, det gør det væsentlig mere læsbart.
Sanitering af inputs spiller. Regex check på både server side og client side er også en god ting, såfremt brugeren skulle slå javascript fra. 

AJAX kaldet på søge funktionen er super, AJAX kaldet på user listen, virker mere som et ekstra unødvendigt trin.

aleksander-GD committed:
Refactored for the review criteria.

aleksander-GD commented:
Jeg har rettet det ift. overflow-wrap.
Jeg har også lavet en nav.css og background.css der bliver importeret i de css filer der har med alle sider at gøre når man er logget ind.

nc-hela committed:
Create Week6Experiments.html

aleksander-GD committed:
Assignment 1 last update

Added background color for the search input field.

aleksander-GD committed:
Updated migration file

Renamed tables to respect the convention. 
Edited DBController to reflect these changes.

nc-hela committed:
mvc exercise

nc-hela committed:
Update router for case sensitivity

nc-hela committed:
Made an api that exposes a list of users with no security

aleksander-GD committed:
Update on branch, not pushed yet

Fixed structure.
Added lektion_6 folder and its exercises

nc-hela committed:
Create Week7Apis.php

aleksander-GD committed:
Update

Exercises for lecture 7 added

aleksander-GD committed:
Merge branch 'Exercises' into aldus17

aleksander-GD committed:
Assignment 2 update

Added the structure.
TODO: begin moving functionality to MVC.

aleksander-GD committed:
Assignment 2 update

Moved functionality and stylying to MVC, this still needs work and understanding.
Added README.md to both assignment 1 and 2.

nc-hela committed:
Create Week7OwnApi.php

aleksander-GD committed:
Assignment 2 update

Added login and registration to MVC, which now works.
Created UserController for user functionality. 
Updated README.
Update Migration file.
Edited image and user model to work with MVC.
Added Controllers to restriced.
Created 3 api, this still needs work.

aleksander-GD committed:
Assignment 2 update

Upload works.
TODO: imagefeed and userlist.

aleksander-GD committed:
Assignment 2 update

Added 2 GET and 1 POST Api's. 
Userlist updated.
All functionality has been moved to MVC and works besides imagefeed.

aleksander-GD committed:
Assignment 2 update

Fixed upload response.

aleksander-GD committed:
Assignment 2 update

CSS added to the pages.
Imagefeed works, but search does not work.
API test on users work, need to finish the POST method so it is correct
Fixed CSS issue on login page

aleksander-GD committed:
Assignment 2 update

Added restriction functionality.
Upload needs bug fixing for display of error and confirmation message
Imagefeed also needs bug fixing, else a service can be made with exactly same functionality as in assignment 1 and use it in the ajax call.

nc-hela committed:
Create Week9Bootstrap.html

nc-hela committed:
Week9Exercises added

nc-hela committed:
Bootstrap exercises

aleksander-GD committed:
Merge branch 'Exercises' into aldus17

aleksander-GD committed:
Assignment 2 update

Fixed githubs/gitkranken shit making

aleksander-GD committed:
Assignment 2 update

Added response check for upload and an image preview.
Imagefeed is done and works.
Added .gitignore for .idea.

nc-hela committed:
Create test.php

aleksander-GD committed:
Assignment 2 update

Added bootstrap and jquery functionality to login, register and upload service.

Redone some of the CSS in the upload section.
Updated the upload section functionality, where you now can better preview the image to be uploaded, you can drag and drop, and remove the image from being uploaded. 

Moved HTML echos from the register endpoint to the RegisterService.

LoginService created, needs to be done for the login
HTML echos as in RegisterService.

Added hover animation for forms.

Added bootstrap alerts for regexChecks on the server side. 

Tested that API post and get works. Updated Post method in regards to how its contents are read from the json post.

aleksander-GD committed:
Assignment 2 update

Added bootstrap and jquery functionality to login, register and upload service.

Redone some of the CSS in the upload section.
Updated the upload section functionality, where you now can better preview the image to be uploaded, you can drag and drop, and remove the image from being uploaded. 

Moved HTML echos from the register endpoint to the RegisterService.

LoginService created, needs to be done for the login
HTML echos as in RegisterService.

Added hover animation for forms.

Added bootstrap alerts for regexChecks on the server side.

aleksander-GD committed:
Merge remote-tracking branch 'origin/aldus17' into aldus17

aleksander-GD committed:
updated json file, to large blob.

aleksander-GD committed:
Merge branch 'Exercises' into aldus17

aleksander-GD committed:
Assignment 2 update

Edited Api controller to satisfy test.php. The test.php file passed on all APIs.

aleksander-GD committed:
Assignment 2 update 

Began progress on moving html echos to service and putting the data responsibility on model level e.g. post sanitation.

aleksander-GD committed:
Assignment 2 update

Updated Api controller, moved out the post in regards to pictures
Updated home controller moved out the post in regards to register and login.
Updated loginService to reflect home controller changes.
Updated registerService to reflect home controller changes.
Updated user controller

Updated model layer for both User and Image to reflect above changes.

aleksander-GD committed:
Assignment 2 update 

Moved Post and echos from upload endpoint to the uploadService
Fixed the navbar
Added smoth navbar hiding on scroll.
Added icons to the navbar items
Updated login endpoint
Added ajax loader to the userlist
Added newer version of bootstrap, jquery and more.

aleksander-GD committed:
Assignment 2 update

Updated ajax call to use jquery for images. 
Added loading indicator when search is made for pictures.
Tested for XSS and no breaches were found.

aleksander-GD committed:
Assignment 2 done

Fixed a issue on the registration page.

aleksander-GD commented:
Assignment 2 færdig. 
Det er klar til review ;) @JacobGlumby1

aleksander-GD committed:
update
